checker/src/main/java/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -74,11 +73,69 @@ protected void addComputedTypeAnnotations(
         // but adding the top type to them, which permits flow-sensitive type refinement.
         // (When it is possible to default types based on their TypeKinds,
         // this whole method will no longer be needed.)
+        addSignednessAnnotation(tree, type);
         addUnknownSignednessToSomeLocals(tree, type);
 
         super.addComputedTypeAnnotations(tree, type, iUseFlow);
     }
 
+    private void addSignednessAnnotation(Tree tree, AnnotatedTypeMirror type) {
+        TypeMirror javaType = type.getUnderlyingType();
+        TypeKind javaTypeKind = javaType.getKind();
+
+        switch (tree.getKind()) {
+            case IDENTIFIER:
```

Comment:
Why is this restricted to just two tree kinds?  Document the reason, or expand it.

Reply:
The whole point of this pull request was *not* to special-case identifiers but make the code more general.
This pull request does not seem to achieve that.  It just slightly expands the special case.
I think this logic should apply to all expressions.  Please also add tests for other types of expressions (for example, arithmetic operations.)